test/e2e/ipsec.go

```diff
@@ -0,0 +1,160 @@
+				gomega.Expect(metricsOutput).Should(gomega.ContainSubstring(targetMetrics))
+				ginkgo.By(fmt.Sprintf("Verify ipsec tunnel metrics reflecting up from node %s. ", nodeInfo.name))
+				gomega.Expect(metricsOutput).Should(gomega.ContainSubstring(ipsecTunnelMetrics[0]))
+			}
+
+			ipsecContainer := "ovn-ipsec"
+			ginkgo.By(fmt.Sprintf("Kill pluto process in ipsec pod which was deployed on node %s.", node1.name))
+			ipsecPod, err := f.ClientSet.CoreV1().Pods(ovnKubernetesNamespace).List(context.TODO(), metav1.ListOptions{
+				LabelSelector: "app=ovn-ipsec",
+				FieldSelector: "spec.nodeName=" + node1.name,
+			})
+			if err != nil {
+				framework.Failf("could not get ipsec pods: %v", err)
+			}
+
+			_, err = e2ekubectl.RunKubectl(ovnKubernetesNamespace, "exec", ipsecPod.Items[0].Name, "--container", ipsecContainer, "--",
+				"bash", "-c", "sudo pkill pluto")
+			framework.ExpectNoError(err, "killing pluto process failed")
+			time.Sleep(10 * time.Second)
+
+			for _, nodeInfo := range nodesIPinfo {
+				metricsOutput := getIPSecMetricsFromNode(f, nodeInfo.name)
+
+				ginkgo.By(fmt.Sprintf("Verify ipsec tunnel metrics reflecting down from node %s. ", nodeInfo.name))
+				gomega.Expect(metricsOutput).Should(gomega.ContainSubstring(ipsecTunnelMetrics[1]))
+			}
+
+			// pluto process will be back automatically.
+			ginkgo.By(fmt.Sprintf("Wait pluto process back on node %s", node1.name))
+			gomega.Eventually(func() bool {
+				output, _ := e2ekubectl.RunKubectl(ovnKubernetesNamespace, "exec", ipsecPod.Items[0].Name, "--container", ipsecContainer, "--",
+					"bash", "-c", "sudo ps -ef |grep pluto")
+				return strings.Contains(output, " /usr/libexec/ipsec/pluto --leak-detective --config /etc/ipsec.conf")
+			}, 120*time.Second, 10*time.Second).Should(gomega.BeTrue())
+
+			ginkgo.By("Verify ipsec metrics reflecting the up tunnel for all nodes")
+			//Wait for the metrics reflecting the tunnel up for all the nodes
+			gomega.Eventually(func() bool {
+				for _, nodeInfo := range nodesIPinfo {
+					metricsOutput := getIPSecMetricsFromNode(f, nodeInfo.name)
+
+					ginkgo.By(fmt.Sprintf("Verify ipsec tunnel metrics reflecting up again from node %s.", nodeInfo.name))
+					if !strings.Contains(metricsOutput, ipsecTunnelMetrics[0]) {
+						return false
+					}
+				}
+				return true
+			}, 240*time.Second, 30*time.Second).Should(gomega.BeTrue())
+		})
+
+	})
+
+})
+
+// getIPSecMetricsFromNode retrieves IPSec metrics from the specified node
+func getIPSecMetricsFromNode(f *framework.Framework, nodeName string) string {
+	ovnKubernetesNamespace := deploymentconfig.Get().OVNKubernetesNamespace()
+	containerName := "ovnkube-node"
+	if isInterconnectEnabled() {
+		containerName = "ovnkube-controller"
+	}
+
+	ovnKubeNodePods, err := f.ClientSet.CoreV1().Pods(ovnKubernetesNamespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "name=ovnkube-node",
+		FieldSelector: "spec.nodeName=" + nodeName,
```

Comment:
this need not filter based on node as we need to check the metrics on all nodes.

Reply:
rewrite it to check metrics from all ovn pods